cmd/soroban-cli/src/rpc/mod.rs

```diff
@@ -216,7 +216,11 @@ pub struct SimulateTransactionResponse {
     pub results: Vec<SimulateHostFunctionResult>,
     #[serde(rename = "transactionData")]
     pub transaction_data: String,
-    #[serde(deserialize_with = "deserialize_default_from_null")]
+    #[serde(
```

Comment:
@willemneal , this fixed the problem with `events` for sim on contract install, now, when the e2e tests run they're hitting similar problem on `minResourceFee` for sim on contract install:

```
running command soroban [contract deploy --wasm-hash ae3b21c20f9ddd129a785b361df1f01687a87069566134192af1fde422cd2a94 --rpc-url http://localhost:8000/soroban/rpc --source SC5O7VZUXDJ6JBDSZ74DSERXL7W3Y5LTOAMRF7RQRL3TAGAPS7LUVG3L --network-passphrase Standalone Network ; February 2017] 

error: jsonrpc error: Parse error: missing field `minResourceFee` at line 1 column 256
running command rm [-rf test_tmp_workspace] 

```

this is the sim json response from rpc server I captured in debug logs, looks like `minResourceFee` is only embedded under transactionData instances, not at top level as mapping expects?

```
soroban-rpc  | time="2023-08-23T22:50:51.645Z" level=debug msg="finished JSONRPC request result" duration=2.570792ms json_req=8 pid=1031 req=8 result="{\"transactionData\":\"AAAAAAAAAAAAAAABAAAAB07Iani5EEiWT7JQSHl/P+i4/cvxxH9D+TlUp+qH3isiAAAAAAABZEcAAAAoAAACgAAAAAAAAAAAAAAACw==\",\"minResourceFee\":\"33855\",\"results\":[{\"auth\":null,\"xdr\":\"AAAADQAAACBOyGp4uRBIlk+yUEh5fz/ouP3L8cR/Q/k5VKfqh94rIg==\"}],\"cost\":{\"cpuInsns\":\"41207\",\"memBytes\":\"3108\"},\"restorePreamble\":{\"transactionData\":\"\",\"minResourceFee\":\"0\"},\"latestLedger\":\"100\"}" status=ok subsys=jsonrpc
```